saltgui/static/scripts/stringify.js

```diff
@@ -0,0 +1,32 @@
+class Stringify {
+
+  // format a JSON object
+  // currently using the builtin function
+  // with small improvement for empty dict or array
+  static format(a) {
+
+    // first use a simple flat format
+    let txt = JSON.stringify(a);
+
+    if(txt === "[]") {
+      // show the brackets a bit wider apart
+      // no support for nested occurances yet
+      txt = "[&nbsp;]";
+    }
+    else if(txt === "{}") {
+      // show the brackets a bit wider apart
+      // no support for nested occurances yet
+      txt = "{&nbsp;}";
+    }
+    else {
+      // use a multi-line format with indentation
+      txt = JSON.stringify(a, null, "  ");
+    }
+
+    return txt;
+  }
+
+}
+
+// for unit tests
+if(typeof module !== "undefined") module.exports = Stringify;
```

Comment:
I've also noticed this in the documentation class. Isn't this something we can avoid?

Reply:
As the comment says, you need this for the unit tests.
This is basically the equivalent of the `<script ... >` statement from `index.html`.